### ISSUE TITLE
Y24-219 - Link tag groups with tag sets

### ISSUE THREAD

**Issue by yoldas** [sanger/sequencescape#4237]
**User story**
As PSD, we would like to create TagSet records and link them with the existing TagGroup records.

**Who are the primary contacts for this story**
Abdullah, Andrew, Katy, Neil

**Who is the nominated tester for UAT**
PSD

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] In Sequencescape databases, of local, UAT, Training environments, extract a list tag groups that belong to single index tag sets and extract a list of **pairs** of tag groups that belong to dual index tag sets. [Obsolete; See the update below]
- [ ] Using record loader files, or rake tasks (whichever is suitable for the environment), create tag set records while linking them with the above. For example, 
    - Single index tag group with name `PN-1000212 Single Index Kit N, Set A` can be linked with the tag set with the same name, while setting `tag_group_id`. [Obsolete; See the update below]
    - Dual index tag groups with names `PN-1000215/PN-3000431 Dual Index Kit TT, Set A (i7)` and `PN-1000215/PN-3000431 Dual Index Kit TT, Set A (i5)` can be linked with the tag set with the name without the specific index suffix `PN-1000215/PN-3000431 Dual Index Kit TT, Set A`, while setting `tag_group_id` and `tag2_group_id` respectively.

Update after review session:

- [ ] Provide a way to link only the following tag groups for this story: 
  - PN-1000215/PN-3000431 Dual Index Kit TT, Set A
    - PN-1000215/PN-3000431 Dual Index Kit TT, Set A (i7)
    - PN-1000215/PN-3000431 Dual Index Kit TT, Set A (i5)
  - PN-1000250 Dual Index Kit TN, Set A
    - PN-1000250 Dual Index Kit TN, Set A (i7)
    - PN-1000250 Dual Index Kit TN, Set A (i5)
  - PN-1000251 Dual Index Kit TS, Set A
    - PN-1000251 Dual Index Kit TS, Set A (i7)
    - PN-1000251 Dual Index Kit TS, Set A (i5)
 - [ ] The solution or the approach should be simple and reusable, more will be linked later to support initial use. There is also a separate story for users to do these themselves already in the future.
 - [ ] Create a new adapter type with name 'Dual Index' and assign the adapter types of both tag groups to this new adapter type; hence tag set shows the first as its own. [Obsolete; not needed for this story]

**Dependencies**
This story is blocked by the following dependencies:
- #4160
- #4236

References
This story has a non-blocking relationship with:
See links in the parent research story.

**Additional context**
Note that we are not renaming any existing tag groups; we are creating new tag sets and links to existing tag groups.

**Comment by KatyTaylor** [comment]
Ideally this would be done after https://github.com/sanger/sequencescape/issues/4239, so that no new tag groups can be created that don't have tag sets.

But probably doesn't matter too much because I don't think new tag groups are created v often.

**Comment by yoldas** [comment]
> Ideally this would be done after #4239, so that no new tag groups can be created that don't have tag sets.
> 
> But probably doesn't matter too much because I don't think new tag groups are created v often.

I have added this story as a direct dependency in https://github.com/sanger/sequencescape/issues/4083 and https://github.com/sanger/limber/issues/1814 with the same idea. 

I had a similar assumption about new tag groups not being created often, which would require the SS tag creation story if done in UI. Record loader base additions for new tag sets and tag groups can be done easily in a similar way after this story I think.
